tests/cases/conformance/controlFlow/controlFlowIterationErrorsAsync.ts

```diff
@@ -0,0 +1,137 @@
+// @strict: true
```

Comment:
I basically just copy-pasted existing [controlFlowIterationErrors](https://github.com/microsoft/TypeScript/blob/d234453328408dd1bbc7858db428b482bfcf0c6b/tests/cases/conformance/controlFlow/controlFlowIterationErrors.ts) test case and I converted all of the functions to be async.

In reality, only [the last two cases](https://github.com/microsoft/TypeScript/blob/d234453328408dd1bbc7858db428b482bfcf0c6b/tests/cases/conformance/controlFlow/controlFlowIterationErrors.ts#L74-L93) changed behavior with this PR so maybe I should limit what had been added to this test case? Either way, happy to clean up if needed - for now I've left those alone.